code/lib/cli/src/scaffold-new-project.ts

```diff
@@ -0,0 +1,180 @@
+import boxen from 'boxen';
+import chalk from 'chalk';
+import prompts from 'prompts';
+import dedent from 'ts-dedent';
+import { logger } from '@storybook/node-logger';
+import { GenerateNewProjectOnInitError } from '@storybook/core-events/server-errors';
+
+import type { PackageManagerName } from './js-package-manager';
+import { exec } from './utils';
+
+type CoercedPackageManagerName = 'npm' | 'yarn' | 'pnpm';
+
+interface SupportedProject {
+  displayName: {
+    type: string;
+    builder?: string;
+    language: string;
+  };
+  createScript: Record<CoercedPackageManagerName, string>;
+}
+
+/**
+ * The supported projects.
+ */
+const SUPPORTED_PROJECTS: Record<string, SupportedProject> = {
+  'react-vite-ts': {
+    displayName: {
+      type: 'React',
+      builder: 'Vite',
+      language: 'TS',
+    },
+    createScript: {
+      npm: 'npm create vite@latest --yes . -- --template react-ts',
+      yarn: 'yarn create vite@latest --yes . --template react-ts',
+      pnpm: 'pnpm create vite@latest --yes . --template react-ts && pnpm i --prefer-offline',
```

Comment:
If this runs actually on the user's system, we should ditch the `--prefer-offline` flag. It was just introduced for performance reasons on CI to generate the sandboxes.